### PR TITLE
Extend DpShutdown Test, PerlDDS Improvements

### PR DESCRIPTION
**Pull request by iguessthislldo**
- DpShutdown Test
  - Extend to cover shutdown and delete related stuff added in https://github.com/objectcomputing/OpenDDS/pull/3159
  - Reformat to bring in line with the style guidelines
  - Cleanup `run_test.pl`
- PerlDDS
  - Remove unused `value_flag` and `get_value_flag`, add ability to get flag value using an optional argument to `flag`.
  - Add support for `DCPSLogLevel`
  - Put `-ORBLogFile` first to make sure debug level set and other messages are in the log file.
- Fix typos in log messages

### CODE REVIEW

File: tests/DCPS/DpShutdown/dpshutdown.cpp [hunk 1]
```diff
@@ -1,322 +1,395 @@
...
+    if (!pub) {
+      ACE_ERROR_RETURN((LM_ERROR, "%N:%l: main() ERROR: create_publisher failed!\n"), 1);
+    }
 
-      // Create another transport instance for participant2 since RTPS transport instances
-      // cannot be shared by domain participants.
-      OpenDDS::DCPS::TransportConfig_rch config2 =
-        OpenDDS::DCPS::TransportRegistry::instance()->get_config(config_2.c_str());
+    // Create DataWriter
+    DDS::DataWriter_var dw = pub->create_datawriter(topic1.in(),
```
Comment (mitza-oci) [inline tests/DCPS/DpShutdown/dpshutdown.cpp:183]:
```suggestion
    DDS::DataWriter_var dw = pub->create_datawriter(topic1,
```

If we're going to "re-style" I'd suggest taking out `.in()` everywhere it's not needed, making `_nil()` just `0` and not use `DEFAULT_STATUS_MASK` when there's no listener.

Certainly not required though.

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: bin/PerlDDS/Run_Test.pm [hunk 2]
```diff
@@ -676,8 +644,7 @@ sub process {
     $file_name =~ s/ /_/g;
     $file_name =~ s/#//g;
 
-    my $debug = " -ORBLogFile $file_name.log";
-    $params .= $debug;
+    $params = "-ORBLogFile $file_name.log $params"
```
[the remark this answers is not in the document]
Comment (iguessthislldo) [reply bin/PerlDDS/Run_Test.pm:647]: Why doesn't this cause a syntax error?

[the remark this answers is not in the document]
Comment (mitza-oci) [reply bin/PerlDDS/Run_Test.pm]: https://perldoc.perl.org/perlsyn#Simple-Statements

**Review by mitza-oci — APPROVED** [review]
Are the errors in tests/DCPS/MultiTopic/run_test.pl due to changes in bin/PerlDDS/Run_Test.pm?

**Pull request merged**